kbase-extension/static/kbase/js/kbaseNarrative.js

```diff
@@ -701,59 +790,76 @@ define([
      *    deletion, etc.
      * 2. Initializes the Core UI dialogs that depend on configuration information (About,
-     *    Upgrade, and Shutdown)
-     * 3. Initializes the
+     *    Upgrade, and Shutdown).
+     * 3. Initializes the help tour.
+     *
+     * The rest depends on a few Jupyter events being fired. Once the notebook is registered
+     * as "loaded" (notebook_loaded.Notebook), we can proceed to load the data, apps, and
+     * side panel components.
+     *
+     * When the kernel is connected (the channel between front and back ends, with the
+     * kernel_connected.Kernel event), we can set up the job communication channel.
+     *
+     * Since these are handled by jquery events, we need an optional callback. The
+     * jobsReadyCallback function is invoked after setting up (or failing to set up) the
+     * job communication channel with the kernel. It takes an (optional) error object as
+     * input, which will have structure { error: xxx }, where xxx is the structure of the error
+     * that might get thrown by the Jupyter stack.
+     *
      */
-    // This should not be run until AFTER the notebook has been loaded!
-    // It depends on elements of the Notebook metadata.
-    Narrative.prototype.init = function () {
+    Narrative.prototype.init = function (jobsReadyCallback) {
         // NAR-271 - Firefox needs to be told where the top of the page is. :P
         window.scrollTo(0, 0);
 
         this.authToken = NarrativeLogin.getAuthToken();
         this.userId = NarrativeLogin.sessionInfo.user;
 
-        Jupyter.narrative.patchKeyboardMapping();
+        this.patchKeyboardMapping();
         this.registerEvents();
         this.initAboutDialog();
         this.initUpgradeDialog();
         this.initShutdownDialog();
         this.initTour();
 
         /* Clever extension to $.event from StackOverflow
-         * Lets us watch DOM nodes and catch when a widget's node gets nuked.
-         * http://stackoverflow.com/questions/2200494/jquery-trigger-event-when-an-element-is-removed-from-the-dom
-         *
-         * We bind a jQuery event to a node. Call it 'destroyed'.
-         * When that event is no longer bound (i.e. when the node is removed, OR when .unbind is called)
-         * it triggers the 'remove' function. Lets us keep track of when widgets get removed
-         * in the registerWidget function below.
-         */
+        * Lets us watch DOM nodes and catch when a widget's node gets nuked.
+        * http://stackoverflow.com/questions/2200494/jquery-trigger-event-when-an-element-is-removed-from-the-dom
+        *
+        * We bind a jQuery event to a node. Call it 'destroyed'.
+        * When that event is no longer bound (i.e. when the node is removed, OR when .unbind is called)
+        * it triggers the 'remove' function. Lets us keep track of when widgets get removed
+        * in the registerWidget function below.
+        */
         $.event.special.destroyed = {
```

Comment:
I don't think this ever gets invoked, and can probably be removed as well. A later PR.